File: backend/src/main/java/by/fpmibsu/findafriend/controller/commands/places/UpdatePlaceHandler.java
package by.fpmibsu.findafriend.controller.commands.places;

import by.fpmibsu.findafriend.application.mediatr.RequestHandler;
import by.fpmibsu.findafriend.dataaccesslayer.place.PlaceDao;
import by.fpmibsu.findafriend.entity.Place;


public class UpdatePlaceHandler extends RequestHandler<Boolean, UpdatePlaceCommand> {
    private final PlaceDao placeDao;

    public UpdatePlaceHandler(PlaceDao placeDao) {
        this.placeDao = placeDao;
    }

    @Override
    public Boolean handle(UpdatePlaceCommand request) {
        Place place = placeDao.getEntityById(request.placeId);
        if (place == null) {
            return false;
        }
        if (request.country != null) {
            place.setCountry(request.country);
        }
        if (request.city != null) {
            place.setCity(request.city);
        }
        if (request.district != null) {
            place.setDistrict(request.district);
        }
        if (request.region != null) {
            place.setRegion(request.region);
        }
        placeDao.update(place);
        return true;
    }
}

File: BackendNET/Backend/Commands/Places/UpdatePlace/UpdatePlaceHandler.cs
using Backend.Repository;
using MediatR;

namespace Backend.Commands.Places.UpdatePlace;

public class UpdatePlaceHandler : IRequestHandler<UpdatePlaceCommand, bool>
{
    private readonly IPlaceRepository _placeRepository;

    public UpdatePlaceHandler(IPlaceRepository placeRepository)
    {
        _placeRepository = placeRepository;
    }

    public async Task<bool> Handle(UpdatePlaceCommand request, CancellationToken cancellationToken)
    {
        var place = await _placeRepository.GetByIdAsync(request.PlaceId, cancellationToken);
        if (place is null)
        {
            return false;
        }

        place.Country = request.Country ?? place.Country;
        place.Region = request.Region ?? place.Region;
        place.City = request.City ?? place.City;
        place.District = request.District ?? place.District;
        await _placeRepository.UpdateAsync(place, cancellationToken);
        return true;
    }
}